### DISCUSSION TITLE
Global or local IScheduler?

### DISCUSSION THREAD

**Discussion by viveknuna** [quartznet/quartznet#2157] [Q&A]
I have asp.net core api. I’m using quartz to  schedule jobs. We are multi tenant application. For each tenant we have different databases in sql server. So we have separate set of quartz related tables for each tenant in their own database . We have a static dictionary <string, IScheduler> which has tenantid as key and scheduler for each tenant. 

So for each tenant/database there’s different scheduler. We use the same instance of IScheduler throughout the application.

We see lots of issues related to timeout while scheduling jobs, checking job exists.

and many times we see triggers in error state.

sorry I have doubt, shall we have such long life time schedulers? So shall we have short lived schedulers like create the Ischeduler instance at the time of actually scheduling job and dispose it.

**Comment by lahma** [comment]
Usually schedulers are alive for the time the process is running. You can have schedulers for maintenance purposes which are not started and are just there to get the API for DB CRUD operations and then there are separate machines/processes handling the actual work.

It's impossible to guess why you have bottlenecks or timeouts, you need to monitor infra and services, especially database.
